opm/simulators/wells/BlackoilWellModelConstraints.cpp

```diff
@@ -476,11 +476,21 @@ actionOnBrokenConstraints(const Group& group,
                           DeferredLogger& deferred_logger) const
 {
 
-    bool changed = false;
     const Group::ProductionCMode oldControl =
         wellModel_.groupState().production_control(group.name());
-
     std::string ss;
+    if (newControl == Group::ProductionCMode::FLD && oldControl != Group::ProductionCMode::FLD) {
+        group_state.production_control(group.name(), newControl);
+        ss = fmt::format("Switching production control mode for group {} from {} to {}",
+        group.name(),
+        Group::ProductionCMode2String(oldControl),
+        Group::ProductionCMode2String(newControl));
+        if (!ss.empty() && wellModel_.comm().rank() == 0)
```

Comment:
When would `ss` ever be `.empty()` in this context?  As far as I can tell, this condition is equivalent to `rank() == 0`, so you might as well write this as
```C++
if (newControl ...) {
    if (rank == 0) {
        deferred_logger.debug(fmt::format(...));
    }

    return true;
}
```
in which case the `ss` object need not be introduced until the "exceed action" switch below.

I also think I'd add a comment to this block to explain why the `FLD` handling is a special case.  This, I think, will aid future maintainers.

Reply:
Thanks for the review. I have updated the code accordingly.